crates/binswap-github/src/lib.rs

```diff
@@ -172,8 +178,17 @@ impl BinswapGithub {
             None,
             Duration::from_millis(5),
             NonZeroU64::new(1).unwrap(),
+            None,
         )?;
 
+        let gh_api_client = GhApiClient::new(
+            client.clone(),
+            env::var("GH_TOKEN")
+                .or_else(|_| env::var("GITHUB_TOKEN"))
+                .ok()
+                .map(Into::into),
+        );
```

Comment:
We hit issues with rate limits quite regularly when we used `binswap` at the university network, so being able to add this here is super neat!

Reply:
> We hit issues with rate limits quite regularly when we used `binswap` at the university network, so being able to add this here is super neat!

I also hit that frequently, so I introduced `GhApiClient` which dramatically improve the situation by using Restful API or GraphQL API (only if a token is provided) of GitHub.

NOTE that unauthorized users also subject to harsh rate limit in GitHub, so providing a GitHub token is recommended.